src/components/app/data/hooks/useEnterpriseFeatures.test.jsx

```diff
@@ -1,65 +1,109 @@
 const mockEnterpriseCustomer = enterpriseCustomerFactory();
 const mockAuthenticatedUser = authenticatedUserFactory();
 const mockEnterpriseLearnerData = {
   enterpriseCustomer: mockEnterpriseCustomer,
-  enterpriseCustomerUserRoleAssignments: [],
   activeEnterpriseCustomer: null,
   activeEnterpriseCustomerUserRoleAssignments: [],
   allLinkedEnterpriseCustomerUsers: [],
   staffEnterpriseCustomer: null,
+  enterpriseFeatures: {
+    isBFFEnabled: false,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+};
+
+const mockBFFDashboardData = {
+  enterpriseCustomer: mockEnterpriseCustomer,
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
 };
-const mockEnterpriseFeatures = mockEnterpriseLearnerData.enterpriseFeatures;
 
+const mockExpectedEnterpriseFeatures = (isMatchedRoute) => (isMatchedRoute
+  ? mockBFFDashboardData.enterpriseFeatures
+  : mockEnterpriseLearnerData.enterpriseFeatures);
 describe('useEnterpriseFeatures', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ initialEntries = [], children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={initialEntries}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          <Routes>
+            <Route path=":enterpriseSlug" element={children} />
+            <Route path=":enterpriseSlug/search" element={children} />
+          </Routes>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
 
   beforeEach(() => {
     jest.clearAllMocks();
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
   });
 
   it.each([
-    { hasQueryOptions: false },
-    { hasQueryOptions: true },
-  ])('should return enterprise features correctly (%s)', async ({ hasQueryOptions }) => {
+    { isMatchedRoute: false },
+    { isMatchedRoute: true },
+  ])('should return enterprise features correctly (%s)', async ({ isMatchedRoute }) => {
     const mockSelect = jest.fn(data => data.transformed);
+    const initialEntries = isMatchedRoute ? ['/test-enterprise'] : ['/test-enterprise/search'];
     const { result, waitForNextUpdate } = renderHook(
       () => {
-        if (hasQueryOptions) {
+        if (isMatchedRoute) {
           return useEnterpriseFeatures({ select: mockSelect });
```

Comment:
similar comment here around whether we need to test `mockSelect` on non-BFF matched routes.